### PR TITLE
[boost] get rid of compiler version in user config

### PR DESCRIPTION
**Pull request by SSE4**
closes: #1084

Specify library name and version:  **boost/all**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SSE4** [comment]
few things:
- MSVC might still need compiler version to select proper one (in case of multiple installations), I am checking this
- going to test on Ubuntu (as our CI didn't catch the original problem) and will report back here

**Comment by jgsogo** [comment]
Waiting for your report, @SSE4 , but the recipe looks great with these changes

**Comment by SSE4** [comment]
1. for Ubuntu, I have successfully reproduced the problem on **ubuntu:18.04** docker image. the PR fixes the problem for me.
2. for MSVC, I am going to push an additional commit - it seems to be common recommendation to specify a version for MSVC toolset.

**Comment by SSE4** [comment]
ready to merge now, should be much simpler and correct than before

File: recipes/boost/all/conanfile.py [hunk 1]
```diff
@@ -760,6 +760,18 @@ def create_library_config(name):
...
         tools.save(filename,  contents)
 
+    @property
+    def _toolset_version(self):
+        if self._is_msvc:
+            compiler_version = str(self.settings.compiler.version)
+            if Version(compiler_version) >= "16":
+                return "14.2"
+            elif Version(compiler_version) >= "15":
+                return "14.1"
```
Comment (jgsogo) [inline recipes/boost/all/conanfile.py]: Should this match the actual toolset in the Conan profile?

Comment (SSE4) [reply recipes/boost/all/conanfile.py]: to be honest, no idea (yet). that's the same like it was in previous version (so far, nobody complained)

**Comment by SSE4** [comment]
I am not sure what was an error... error log is obscure

**Comment by SSE4** [comment]
okay, the missing part was ".0" for MSVC <= 14, now it's building

**Comment by SSE4** [comment]
@danimtb  @jgsogo any idea on unexpected error(s)?

**Comment by danimtb** [comment]
Related to #1020

**Comment by danimtb** [comment]
Fix has been deployed. Relaunching the CI for this!

**Pull request merged**